### ISSUE TITLE
I want to use the Azure open ai knowledge base. Currently it's only giving the results from the PDF. I want results outside the PDF. Thanks in advance.

### ISSUE THREAD

**Issue by rafat-git** [Azure-Samples/azure-search-openai-demo#1632]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
>

### Any log messages given by the failure
>

### Expected/desired behavior
>

### OS and Version?
> Windows 7, 8 or 10. Linux (which distribution). macOS (Yosemite? El Capitan? Sierra?)

### azd version?
> run `azd version` and copy paste here.

### Versions
>

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by SethJonesOSA** [comment]
> I too often think about RAG solutions having a '3 mode switch':
> 
> * 1: Pure RAG: Only use the content I provide below.
> * 2: Blend: Content below + your own training to enrich or challenge.
> * 3: Web Blend: Use web-search from a trusted domain to enrich or challenge.
> 
> Maybe #2 isn't needed. I then get lost in an endless loop of asking myself why the user should have to direct the model to behave according to a toggle. Seems like a 'blame the victim' development error to me.
> 
> Perhaps the best answer is #3, where the model sublimely uses your in-content/index-based data and uses a google/bing api to search/snippet fact check; perhaps presenting the split screen result with the "Based on your data" on the left and "Based on my enrichment" on the right.
> 
> Would be interesting to read a blog about this, as I'm sure folks at Glean, Cohere and other RAG shops wrestle with this. @pamelafox is, by far, closer to this than I. Would love her pov.

I think what you're looking for is "agentic ai". Currently this does not exist within this framework but there are numerous resources that explain this architecture.

**Comment by evogelpohl** [comment]
> > I too often think about RAG solutions having a '3 mode switch':
> > 
> > * 1: Pure RAG: Only use the content I provide below.
> > * 2: Blend: Content below + your own training to enrich or challenge.
> > * 3: Web Blend: Use web-search from a trusted domain to enrich or challenge.
> > 
> > Maybe #2 isn't needed. I then get lost in an endless loop of asking myself why the user should have to direct the model to behave according to a toggle. Seems like a 'blame the victim' development error to me.
> > Perhaps the best answer is #3, where the model sublimely uses your in-content/index-based data and uses a google/bing api to search/snippet fact check; perhaps presenting the split screen result with the "Based on your data" on the left and "Based on my enrichment" on the right.
> > Would be interesting to read a blog about this, as I'm sure folks at Glean, Cohere and other RAG shops wrestle with this. @pamelafox is, by far, closer to this than I. Would love her pov.
> 
> I think what you're looking for is "agentic ai". Currently this does not exist within this framework but there are numerous resources that explain this architecture.

Indeed. Love AutoGen and others out there, use them often. Tho, Chat w/ Your Data solutions have become a thing in terms of a product definition. Having the prompt demand that the model to "only comment on what's presented below and say I don't know if not..." seems like a stop-gap architecture. 

I'm casting no shade on this project. It's an outstanding application of the art. More, a comment on the 'pure rag' pattern overall. I just don't see it scaling & providing rich/complete narratives if we keep the prompt so narrowly confined.
